src/linear_algebra.jl

```diff
@@ -127,9 +127,14 @@ function mutable_operate!(
     return _mutable_operate!(add_sub_op(op), A, B, (α1, α2), β)
 end
 
-# Fallback, we may be able to be more efficient in more cases by adding more specialized methods
+# Fallback, we may be able to be more efficient in more cases by adding more
+# specialized methods.
 function mutable_operate!(op::AddSubMul, A::Array, x, y, args::Vararg{Any,N}) where {N}
-    return mutable_operate!(op, A, x, *(y, args...))
+    if N == 0
+        return mutable_operate!(op, A, operate!(*, x, y))
```

Comment:
I'd rather create another method `mutable_operate!(op::AddSubMul, A::Array, x, y)` than an `if` in this one.

Reply:
Is `operate!(*, x, y)` correct here? Or should it be `*(x, y)`

Reply:
No, it should be `*(x, y)`, you are only allowed to modified the first argument.